web/src/hash_util.ts

```diff
@@ -93,7 +94,7 @@ export function search_terms_to_hash(terms?: NarrowTerm[]): string {
 
         for (const term of terms) {
             // Support legacy tuples.
-            const operator = term.operator;
+            const operator = util.canonicalize_stream_synonym(term.operator);
```

Comment:
This may not be required, in that I believe all callers are the output of `filter.terms()`, which has already canonicalized this? Oh, or maybe it is because we're using `stream` in the URLs and `channel` in the internals, in the temporary state? OK.